server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -431,6 +451,25 @@ public void waitForTaskCompletion(Task task, long untilInNanos) {
             this.task = task;
         }
 
+        public void registerChildNode(DiscoveryNode node) {
+            synchronized (this) {
+                if (banChildren) {
+                    throw new TaskCancelledException("The parent task was cancelled, shouldn't start any children tasks");
+                }
+                nodes.add(node);
+            }
+        }
+
+        public Set<DiscoveryNode> startBan() {
+            synchronized (this) {
+                if (banChildren) {
+                    throw new TaskCancelledException("The parent task was cancelled, shouldn't start any children tasks");
```

Comment:
I wonder why we are throwing the exception here. Should this not just ignore if the flag is already set?